core/src/main/java/com/linecorp/armeria/server/DecodedHttpRequest.java

```diff
@@ -99,13 +102,26 @@ protected EventLoop defaultSubscriberExecutor() {
 
     @Override
     public boolean tryWrite(HttpObject obj) {
-        final boolean published = super.tryWrite(obj);
-        if (published && obj instanceof HttpData) {
-            final int length = ((HttpData) obj).length();
-            inboundTrafficController.inc(length);
-            assert ctx != null : "uninitialized DecodedHttpRequest must be aborted.";
-            ctx.logBuilder().increaseRequestLength((HttpData) obj);
+        if (receivedTrailers) {
+            ReferenceCountUtil.safeRelease(obj);
+            return false;
+        }
+
+        boolean published = false;
+        if (obj instanceof HttpHeaders) { // HTTP trailers.
+            published = super.tryWrite(obj);
+            receivedTrailers = true;
```

Comment:
I'm curious if we can just call `super.close()` here rather than having an additional boolean field.

Reply:
That's a good idea. Will fix it.